lib/tests/misc.nix

```diff
@@ -347,6 +347,11 @@ runTests {
     expected = "Hello\\x20World";
   };
 
+  testEscapeQuery = {
+    expr = strings.escapeQuery "Hello World";
```

Comment:
We may want more robust testing for something I expect to have the potential for broad usage. @infinisil how's the property testing lib going?

Reply:
Agreed, added some more tests.